api/dbadapters/bigquery.ts

```diff
@@ -199,13 +200,68 @@ export class BigQueryDbAdapter implements IDbAdapter {
       `SELECT * FROM ${CACHED_STATE_TABLE_NAME}`,
       5000 // TODO: Add pagination for 5000+ rows
     );
-    const persistedMetadata = rows.map(row => {
-      const encodedProto = Buffer.from(row.proto, "base64");
+    const persistedMetadata = rows.map((row: IMetadataRow) => {
+      const encodedProto = Buffer.from(row.metadata_proto, "base64");
       return dataform.PersistedTableMetadata.decode(encodedProto);
     });
     return persistedMetadata;
   }
 
+  public async persistStateMetadata(executionGraph: dataform.IExecutionGraph): Promise<void> {
+    const { actions } = executionGraph;
```

Comment:
i would argue for just passing in an array of `IExecutionAction` objects instead of the whole execution graph